Composer/packages/client/src/pages/botProject/GetAppInfoFromPublishProfileDialog.tsx

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import React, { Fragment, useState, useEffect } from 'react';
+import { jsx } from '@emotion/core';
+import { useRecoilValue } from 'recoil';
+import { PublishTarget } from '@bfc/shared';
+import formatMessage from 'format-message';
+import { DefaultButton, PrimaryButton } from 'office-ui-fabric-react/lib/Button';
+import Dialog, { DialogFooter } from 'office-ui-fabric-react/lib/Dialog';
+import { Dropdown, IDropdownOption } from 'office-ui-fabric-react/lib/Dropdown';
+
+import { settingsState } from '../../recoilModel/atoms/botState';
+
+type AppInfo = {
+  appId: string;
+  appPassword?: string;
+};
+
+type Props = {
+  projectId: string;
+  hidden?: boolean;
```

Comment:
is it needed?

Reply:
What is it? Do you mean should hidden be required?

Reply:
the dialog is rendered conditionally, controlling hidden from outside almost never happens, so it's not needed.